server/fleet/scripts.go

```diff
@@ -373,14 +373,15 @@ type ScriptPayload struct {
 }
 
 type SoftwareInstallerPayload struct {
-	URL               string `json:"url"`
-	PreInstallQuery   string `json:"pre_install_query"`
-	InstallScript     string `json:"install_script"`
-	UninstallScript   string `json:"uninstall_script"`
-	PostInstallScript string `json:"post_install_script"`
-	SelfService       bool   `json:"self_service"`
-	FleetMaintained   bool   `json:"-"`
-	Filename          string `json:"-"`
+	URL                string `json:"url"`
+	PreInstallQuery    string `json:"pre_install_query"`
+	InstallScript      string `json:"install_script"`
+	UninstallScript    string `json:"uninstall_script"`
+	PostInstallScript  string `json:"post_install_script"`
+	SelfService        bool   `json:"self_service"`
+	FleetMaintained    bool   `json:"-"`
+	Filename           string `json:"-"`
+	InstallDuringSetup *bool  `json:"install_during_setup"` // if nil, do not change saved value, otherwise set it
```

Comment:
Why not use `optjson.Bool` here, and similar places?

Reply:
In this case we don't need the extra features of `optjson` (i.e. no need to know if the key was explicitly set or not when it's null). `optjson` is mostly useful for PATCH-like processing where the key being absent is different than the key present but null.